### ISSUE TITLE
Impossible to compile PCL against VTK 6 trunk

### ISSUE THREAD

**Issue by VictorLamoine** [PointCloudLibrary/pcl#1234]
Tested on Ubuntu 14.04 with VTK latest trunk, configuration:
```cmake
 BUILD_DOCUMENTATION              OFF
 BUILD_EXAMPLES                   OFF
 BUILD_SHARED_LIBS                ON
 BUILD_TESTING                    ON
 BUILD_USER_DEFINED_LIBS          OFF
 CMAKE_BUILD_TYPE                 Debug
 CMAKE_INSTALL_PREFIX             /usr/local
 CTEST_TEST_TIMEOUT               3600
 RMANTREE                         RMANTREE-NOTFOUND
 VTK_ANDROID_BUILD                OFF
 VTK_EXTRA_COMPILER_WARNINGS      OFF
 VTK_Group_Imaging                ON
 VTK_Group_MPI                    ON
 VTK_Group_Qt                     ON
 VTK_Group_Rendering              ON
 VTK_Group_StandAlone             ON
 VTK_Group_Tk                     ON
 VTK_Group_Views                  ON
 VTK_Group_Web                    OFF
 VTK_IOS_BUILD                    OFF
 VTK_PYTHON_VERSION               2
 VTK_QT_VERSION                   4
 VTK_RENDERING_BACKEND            OpenGL
 VTK_SMP_IMPLEMENTATION_TYPE      Sequential
 VTK_USE_LARGE_DATA               OFF
 VTK_WRAP_JAVA                    OFF
 VTK_WRAP_PYTHON                  OFF
 VTK_WRAP_TCL                     OFF
```

Point Cloud Library compilation error:
```bash
/usr/bin/ld: cannot find -lvtkRenderingFreeTypeOpenGL
collect2: error: ld returned 1 exit status
```

Maybe there are some directions in the [VTK 6 Migration guide](http://www.vtk.org/Wiki/VTK/VTK_6_Migration_Guide)

**Comment by VictorLamoine** [comment]
I had to change the `CMakeLists.txt` for my VTK (VTK only) test projects; what I don't understand is that the old `CMakeLists.txt` is working fine on other machines with VTK trunk.

```cmake
cmake_minimum_required(VERSION 2.8.7 FATAL_ERROR)
project(test_vtkStripper)
find_package(VTK 6.3 COMPONENTS vtkRenderingOpenGL vtkInteractionStyle vtkRenderingVolumeOpenGL NO_MODULE)
include_directories(${VTK_INCLUDE_DIRS})
set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS ${VTK_DEFINITIONS})
add_executable(test_vtkStripper test_vtkStripper.cpp)
target_link_libraries(test_vtkStripper ${VTK_LIBRARIES})
```

**Comment by mdering** [comment]
seems like I'm also having trouble compiling against the trunk lately. I get a file not found for vtkAtomicTypes.h during compilation. Is this an upstream issue?

**Comment by VictorLamoine** [comment]
The issue I describe is most probably due to changes in VTK CMake configuration.
As for your error, `vtkAtomicTypes.h` is not included anywhere in PCL and I can't find such a file name (neither on my machine, neither on the web)

**Comment by mdering** [comment]
it was pushed as a change [here](https://github.com/Kitware/VTK/commit/3ded35a633a71f344f1078647807c8c3c99e2c5a), seems like it's being left out during the install process, so its likely an upstream problem. All the same, it is a dependency of vtkImageReader2.h which is included in (among other things) image_grabber.cpp.

**Comment by VictorLamoine** [comment]
Looks like it happens when VTK is compiled with `VTK_RENDERING_BACKEND`=OpenGL2.
My first post contradicts this but I must have done something wrong when copying the CMake options.

**Comment by taketwo** [comment]
Should be fixed with #1534.
